RecipeMatch/View Controllers/StreamViewController.m

```diff
@@ -11,41 +11,61 @@
 #import "SceneDelegate.h"
 #import "DraggableViewBackground.h"
 
+@interface StreamViewController()
+@property (nonatomic, strong) NSString *preferences;
+@property (nonatomic, strong) DraggableViewBackground *draggableBackground;
+@end
+
 @implementation StreamViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
     // Do any additional setup after loading the view.
+
+    // add centered logo
+    UIImageView* imageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"logo"]];
+    imageView.contentMode = UIViewContentModeScaleAspectFit;
+
+    UIView* titleView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+    imageView.frame = titleView.bounds;
+    [titleView addSubview:imageView];
+
+    self.navigationItem.titleView = titleView;
     
-    DraggableViewBackground *draggableBackground = [[DraggableViewBackground alloc]initWithFrame:self.view.frame];
-    [self.view addSubview:draggableBackground];
-    
+    UIActivityIndicatorView *spinner = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleMedium];
+    spinner.center = CGPointMake(self.view.center.x, self.view.center.y);
+    spinner.tag = 12;
+    [self.view addSubview:spinner];
+    [spinner startAnimating];
     
-    /*CGRect frame = self.view.frame;
-    frame.origin.y = -self.view.frame.size.height; //optional: if you want the view to drop down
-    DraggableViewBackground *draggableBackground = [[DraggableViewBackground alloc]initWithFrame:frame];
-    draggableBackground.alpha = 0; //optional: if you want the view to fade in
-
-    [self.view addSubview:draggableBackground];
-
-      //optional: animate down and in
-    [UIView animateWithDuration:0.5 animations:^{
-        draggableBackground.center = self.view.center;
-        draggableBackground.alpha = 1;
-    }];
-    */
+    [self showCards];
 }
 
+-(void)showCards{
+    self.draggableBackground = [[DraggableViewBackground alloc]initWithFrame:self.view.frame];
+    self.draggableBackground.preferences = self.preferences;
+    [self.draggableBackground fetchRecipes];
+    
+    [self.view addSubview:self.draggableBackground];
+}
 
+-(void)sendData:(NSString *)prefRequest{
+    if(prefRequest != (id)[NSNull null] && prefRequest.length != 0){
+        self.preferences = prefRequest;
+        [self.draggableBackground removeFromSuperview];
+        [self viewDidLoad];
+    }
+}
 
-/*
 #pragma mark - Navigation
 
 // In a storyboard-based application, you will often want to do a little preparation before navigation
 - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
     // Get the new view controller using [segue destinationViewController].
     // Pass the selected object to the new view controller.
+    PreferencesViewController *preferencesController = [segue destinationViewController];
+    preferencesController.delegate = self; // Set the second view controller's
```

Comment:
same as before, in viewDidLoad you can't expect that the destinationVC is the VC you actually think- you need to think that at some point you'll add more functionality, and might forget that there is another VC here, so unless you protect your self against these kinda cases, the app might crash